File: algorithms/62. Unique Paths/solution3.cc
class Solution {
public:
    int uniquePaths(int m, int n) {
        int dp[101][101] = {0};
        dp[0][1] = 1;
        for ( int i = 1; i <= m; ++i ) {
            for ( int j = 1; j <= n; ++j ) {
                dp[i][j] = dp[i][j-1] + dp[i-1][j];
            }
        }
        return dp[m][n];
    }
};

// 61 / 61 test cases passed.
// Status: Accepted
// Runtime: 3 ms
// beats 2.91 %

File: algorithms/373. Find K Pairs with Smallest Sums/solution3.py
class Solution(object):
    def kSmallestPairs(self, nums1, nums2, k):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :type k: int
        :rtype: List[List[int]]
        """
        # @StefanPochmann
        return map(list, heapq.nsmallest(k, itertools.product(nums1, nums2), key=sum))

# 27 / 27 test cases passed.
# Status: Accepted
# Runtime: 502 ms
# beats 10.60 %
